packages/algob/src/internal/deployer.ts

```diff
@@ -214,14 +237,26 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
     this.cpData.putMetadata(this.networkName, key, value);
   }
 
-  private assertNoAsset (name: string): void {
-    if (this.isDefined(name)) {
-      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
-      throw new BuilderError(
-        ERRORS.BUILTIN_TASKS.DEPLOYER_ASSET_ALREADY_PRESENT, {
-          assetName: name
-        });
-    }
+  /**
+   * Persist checkpoint till current call.
+   */
+  storeCheckpoint (): void {
+    persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+  }
```

Comment:
Also, can we use this function in other places as well then (eg. in `deployASA`) ?

Reply:
yes